app/controllers/assessments_controller.rb

```diff
@@ -111,12 +111,14 @@ def install_assessment
                                          filename)) || (filename == "..") || (filename == ".")
 
       # assessment names must be only lowercase letters and digits
-      if filename =~ /[^a-z0-9]/
+      if filename !~ /^[A-Za-z_][A-Za-z][A-Za-z0-9_-]*$/
         # add line break if adding to existing error message
         flash.now[:error] = flash.now[:error] ? "#{flash.now[:error]} <br>" : ""
         flash.now[:error] += "An error occurred while trying to display an existing assessment " \
-            "from file directory #{filename}: assessment file names must only contain lowercase " \
-            "letters and digits with no spaces"
+            "from file directory #{filename}: Invalid assessment name. "\
+            "Find more information on valid assessment names "\
+            '<a href="https://docs.autolabproject.com/lab/#assessment-naming-rules">here</a><br>'
+        flash[:html_safe] = true
         flash.now[:html_safe] = true
```

Comment:
Only need one of the two, in this case since there's a redirect to install assessment, maybe just flash and remove flash.now.